src/socket.rs

```diff
@@ -660,6 +662,61 @@ impl Socket {
     ) -> io::Result<usize> {
         sys::send_to_vectored(self.as_raw(), bufs, addr, flags)
     }
+
+    /// Sends data on the socket accompanied by ancillary control message data.
+    #[cfg(unix)]
+    pub fn send_msg<B: AsRef<[u8]>>(
+        &self,
+        addr: Option<&SockAddr>,
```

Comment:
Let's split this code into `send_msg` (without an address) and `send_msg_to` (with an address), following the same pattern of the other send functions.